### PR TITLE
Use elfdep To get shlibdeps for DEB

### PR DESCRIPTION
**Pull request by TheBrokenRail**
It also moves the ```tools/makedeb``` script to ```tools/debian/make-deb``` and the docs must be updated accordingly.

### CODE REVIEW

**Comment by TheBrokenRail** [comment]
Note this will fail if any MachO is not the same architecture as the ```clang``` binary, as the ```clang``` binary is used to generate an ELF executable that ```dpkg-shlibdeps``` is run on.

**Comment by TheBrokenRail** [comment]
Just finished building, it works!

**Comment by TheBrokenRail** [comment]
Any updates?

**Review by facekapow — CHANGES_REQUESTED** [review]
Other than that one thing, LGTM and it works nicely (output of `dpkg -I <package>`):
```
 new Debian package, version 2.0.
 size 108558132 bytes: control archive=767948 bytes.
     706 bytes,    10 lines      control
 4446709 bytes, 33299 lines      md5sums
     107 bytes,     5 lines   *  postinst             #!/bin/sh
     107 bytes,     5 lines   *  prerm                #!/bin/sh
 Package: darling
 Version: 0.1.20210221~focal
 Architecture: amd64
 Maintainer: Lubos Dolezel <lubos@dolezel.info>
 Installed-Size: 654769
 Depends: libc6 (>= 2.14), libc6-i386 (>= 2.3), libavcodec58, libavformat58, libavresample4 (>= 7:4.0), libavutil56, libcairo2 (>= 1.2.4), libdbus-1-3 (>= 1.9.14), libegl1, libfontconfig1 (>= 2.12.6), libfreetype6 (>= 2.2.1), libfuse2 (>= 2.2), libgif7 (>= 5.1), libgl1, libglu1-mesa | libglu1, libjpeg8 (>= 2.0), libpng16-16 (>= 1.6.2-1), libpulse0 (>= 0.99.1), libtiff5 (>= 4.0.3), libx11-6, libxcursor1 (>> 1.1.2), libxext6, libxkbfile1 (>= 1:1.1.0), libxrandr2, fuse, darling-dkms
 Section: misc
 Priority: optional
 Description: Darling
  macOS emulation layer for Linux
```

**Review by lastthyme — APPROVED** [review]

File: tools/debian/make-deb [hunk 1]
```diff
@@ -17,4 +17,4 @@ else
     ARG='-b'
 fi
 
-debuild -us -uc ${ARG}
+debuild -us -uc --no-lintian ${ARG}
```
[the remark this answers is not in the document]
Comment (TheBrokenRail) [reply tools/debian/make-deb]: Done!

**Pull request merged**